### PR TITLE
`rustc_session` cleanups

### PR DESCRIPTION
**Pull request by nnethercote**
r? @bjorn3

### CODE REVIEW

File: compiler/rustc_session/src/config.rs [hunk 1]
```diff
@@ -1273,7 +1273,7 @@ fn default_configuration(sess: &Session) -> Cfg {
     }
     ret.insert((sym::target_arch, Some(Symbol::intern(arch))));
     ret.insert((sym::target_endian, Some(Symbol::intern(end.as_str()))));
-    ret.insert((sym::target_pointer_width, Some(Symbol::intern(&wordsz))));
+    ret.insert((sym::target_pointer_width, Some(sym::integer(wordsz))));
```
Comment (bjorn3) [inline compiler/rustc_session/src/config.rs:1276]: This doesn't have any perf improvement (only 0..9 are pre-interned) This code now does have an extra check for wordsz < 10, but given that this only runs once per compiler session, it shouldn't really matter.

File: compiler/rustc_session/src/session.rs [hunk 2]
```diff
@@ -147,7 +145,7 @@ pub struct Session {
     pub target: Target,
     pub host: Target,
     pub opts: config::Options,
-    pub host_tlib_path: Lrc<SearchPath>,
+    host_tlib_path: Lrc<SearchPath>,
     pub target_tlib_path: Lrc<SearchPath>,
```
Comment (bjorn3) [inline compiler/rustc_session/src/session.rs:149]: This feels inconsistent between host_tlib_path and target_tlib_path.

File: compiler/rustc_session/src/session.rs [hunk 3]
```diff
@@ -1725,13 +1723,13 @@ impl EarlyErrorHandler {
...
     #[allow(rustc::diagnostic_outside_of_impl)]
-    pub fn early_note(&self, msg: impl Into<DiagnosticMessage>) {
+    pub(crate) fn early_note(&self, msg: impl Into<DiagnosticMessage>) {
         self.handler.struct_note_without_error(msg).emit()
     }
 
     #[allow(rustc::untranslatable_diagnostic)]
     #[allow(rustc::diagnostic_outside_of_impl)]
-    pub fn early_help(&self, msg: impl Into<DiagnosticMessage>) {
+    pub(crate) fn early_help(&self, msg: impl Into<DiagnosticMessage>) {
```
Comment (bjorn3) [inline compiler/rustc_session/src/session.rs:1729]: Someone out-of-tree might need these.

Comment (nnethercote) [reply compiler/rustc_session/src/session.rs:1729]: I will make these `pub` again: `early_help`, `early_note`, `abort_if_error_and_set_error_format`.

I will also change this one from `pub(crate)` to `pub`, for consistency: `early_struct_error`.

**Comment by nnethercote** [comment]
I updated to address the comments. Thank you for the fast review :)

**Pull request merged**